packages/nhsuk-frontend/src/nhsuk/components/file-upload/file-upload.mjs

```diff
@@ -0,0 +1,492 @@
+import { closestAttributeValue } from '../../common/closest-attribute-value.mjs'
+import { ConfigurableComponent } from '../../configurable-component.mjs'
+import { ElementError } from '../../errors/index.mjs'
+import { I18n } from '../../i18n.mjs'
+
+/**
+ * File upload component
+ *
+ * @preserve
+ * @augments ConfigurableComponent<FileUploadConfig>
+ */
+export class FileUpload extends ConfigurableComponent {
+  /**
+   * @private
+   * @type {HTMLFileInputElement}
+   */
+  $input
+
+  /**
+   * @private
+   */
+  $button
+
+  /**
+   * @private
+   */
+  $status
+
+  /** @private */
+  i18n
+
+  /** @private */
+  id
+
+  /** @private */
+  $announcements
+
+  /**
+   * @private
+   * @type {boolean | undefined}
+   */
+  enteredAnotherElement
+
+  /**
+   * @param {Element | null} $root - File input element
+   * @param {FileUploadConfig} [config] - File Upload config
+   */
+  constructor($root, config = {}) {
+    super($root, config)
+
+    const $input = this.$root.querySelector('input')
+
+    if ($input === null) {
```

Comment:
Interesting that GOV.UK Frontend didn't follow their usual format here to check for incorrect element types. If we can please use:

```suggestion
if (!($input instanceof HTMLInputElement)) {
```